### ISSUE TITLE
Accessibility: The voice message recording button does not respond to Talkback's long click

### ISSUE THREAD

**Issue by RastislavKish** [signalapp/Signal-Android#15028]
### Guidelines

- [x] I have searched open and closed issues for duplicates
- [x] I am submitting a bug report for existing functionality that does not work as intended
- [x] This isn't a feature request or a discussion topic

### Bug description

Talkback has a standard gesture of long pressing a component, which consists of a double tap, where the user holds the second touch as long as they need to keep the icon pressed. However, the Signal's button for recording voice messages does not respond well to this system. currently, it requires the user to perform a double tap, and only afterwards perform a tap and hold gesture to start recording the message. Hence, the process for a blind person consists of 4 touches instead of 2.

## Reproduction steps

1 Open Signal
2. Open any conversation
3. Turn on Talkback
4. Touch the record a voice message button, this will focus it and the screenreader should announce the button's label
5. Perform a tap and hold gesture, i.e. a double tap where the second touch is prolonged for the duration of the voice message

## Expected behavior

The recording should start immediately when long press is detected, and stop when the user releases their finger.

## Actual behavior

Nothing happens. In order to record a voice message, one needs to first make an ordinary double-tap, and then quickly perform a tap and hold, which is inefficient, requires more handwork and is less reliable.

### Device

Xiaomi Redmi note 12 pro 5G, Motorola Moto g 5G plus

### Android version

14, 10

### Signal version

8.27.1, 8.27.1
